data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/config/GetForkSchedule.java

```diff
@@ -13,32 +13,56 @@
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_INTERNAL_ERROR;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_CONFIG;
+import static tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition.listOf;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.http.Context;
-import io.javalin.http.Handler;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.api.ConfigProvider;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.response.v1.config.GetForkScheduleResponse;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.state.Fork;
 
-public class GetForkSchedule implements Handler {
+public class GetForkSchedule extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/config/fork_schedule";
   private final ConfigProvider configProvider;
-  private final JsonProvider jsonProvider;
 
-  public GetForkSchedule(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    this(dataProvider.getConfigProvider(), jsonProvider);
+  private static final SerializableTypeDefinition<List<Fork>> FORK_SCHEDULE_RESPONSE_TYPE =
+      SerializableTypeDefinition.<List<Fork>>object()
+          .name("GetForkScheduleResponse")
+          .description("Retrieve all forks, past present and future, of which this node is aware.")
+          .withField("data", listOf(Fork.getJsonTypeDefinition()), Function.identity())
+          .build();
+
+  public GetForkSchedule(final DataProvider dataProvider) {
+    this(dataProvider.getConfigProvider());
   }
 
-  GetForkSchedule(final ConfigProvider configProvider, final JsonProvider jsonProvider) {
-    this.jsonProvider = jsonProvider;
+  GetForkSchedule(final ConfigProvider configProvider) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getScheduledForks")
+            .summary("Get scheduled forks")
+            .description("Retrieve all scheduled upcoming forks this node is aware of.")
+            .tags(TAG_CONFIG)
+            .response(SC_OK, "Success", FORK_SCHEDULE_RESPONSE_TYPE)
+            .response(SC_INTERNAL_SERVER_ERROR, "Beacon node internal error.")
```

Comment:
ah i think if we remove this internal error listing, then it should become correct in the schema, we just haven't defined the object type here (it's automatically added i think...)